subsys/bluetooth/common/Kconfig

```diff
@@ -8,39 +8,6 @@
 
 if BT_HCI
 
-config BT_HCI_VS
```

Comment:
Initially, BT_HCI_VS was defined in `subsys/bluetooth/common/Kconfig`. But the LL options are applied later in `subsys/bluetooth/controller/Kconfig`. See `subsys/bluetooth/Kconfig`:
```
source "subsys/bluetooth/common/Kconfig"
source "subsys/bluetooth/host/Kconfig"
source "subsys/bluetooth/controller/Kconfig"
source "subsys/bluetooth/shell/Kconfig"
```
In case when LL doesn't support Zephyr's VS commands and events, there is no way to say it, because LL configuration is defined after this configuration.  More over, BT_HCI_VS is enabled by default.

Therefore, in order to control this configuration from LL, this configuration is moved down after the LL configuration. So now LL configuration can say: select BT_HAS_HCI_VS, which would mean that it supports BT_HCI_VS

Reply:
> In case when LL doesn't support Zephyr's VS commands and events, there is no way to say it, because LL configuration is defined after this configuration.

I don't understand how the order can affect things.

My claim is that we can make do with introducing the *HAS* option, we don't need to move things around.

```
It's perfectly valid to do:

config BT_HAS_HCI_VS
	bool

config BT_HCI_VS
	bool "Zephyr HCI Vendor-Specific Commands"
	depends on BT_HAS_HCI_VS
	default y

[...]

select BT_HAS_HCI_VS

```

Reply:
Checked again. True. It isn't needed anymore sinse `BT_HAS_HCI_VS` is added. This code block should be moved back to common/Kconfig together with `BT_HAS_HCI_VS`

Reply:
Fixed.